controllers/imagecollector/imagecollector_controller.go

```diff
@@ -0,0 +1,342 @@
+		ObjectMeta: metav1.ObjectMeta{Name: collectorShared},
+
+		Spec: eraserv1alpha1.ImageCollectorSpec{Images: []eraserv1alpha1.Image{}},
+	}
+
+	if err := r.Get(ctx, types.NamespacedName{Name: collectorShared, Namespace: "default"}, imageCollector); err != nil {
+		if isNotFound(err) {
+			if err := r.Create(ctx, imageCollector); err != nil {
+				log.Info("could not create shared image collector")
+				return reconcile.Result{}, err
+			}
+		} else {
+			log.Info("could not get shared image collector")
+			return reconcile.Result{}, err
+		}
+	}
+
+	imageJobList := &eraserv1alpha1.ImageJobList{}
+	if err := r.List(ctx, imageJobList); err != nil {
+		log.Info("could not list imagejobs")
+		return reconcile.Result{}, err
+	}
+
+	relevantJobs := util.FilterJobListByOwner(
+		imageJobList.Items, metav1.NewControllerRef(imageCollector, schema.GroupVersionKind{
+			Group:   "eraser.sh",
+			Version: "v1alpha1",
+			Kind:    "ImageCollector",
+		}),
+	)
+
+	if len(relevantJobs) > 1 {
+		return reconcile.Result{}, fmt.Errorf("More than one collector ImageJob scheduled")
+	}
+	if len(relevantJobs) == 0 {
+		if res, err := r.createImageJob(ctx, req, imageCollector); err != nil {
+			return res, err
+		}
+		return ctrl.Result{RequeueAfter: repeatPeriod}, nil
+	}
+
+	// else length is 1, so check job phase
+	switch phase := relevantJobs[0].Status.Phase; phase {
+	case eraserv1alpha1.PhaseCompleted:
+		log.Info("completed phase")
+		if relevantJobs[0].Status.DeleteAfter == nil {
+			if res, err := r.updateSharedCRD(ctx, req, imageCollector); err != nil {
+				return res, err
+			}
+			relevantJobs[0].Status.DeleteAfter = util.After(time.Now(), *util.SuccessDelDelaySeconds)
+			if err := r.Status().Update(ctx, &relevantJobs[0]); err != nil {
+				log.Info("Could not update Delete After for job " + relevantJobs[0].Name)
+			}
+		}
+		if res, err := r.handleJobDeletion(ctx, &relevantJobs[0]); err != nil || res.RequeueAfter > 0 {
+			return res, err
+		}
+	case eraserv1alpha1.PhaseFailed:
+		log.Info("failed phase")
+		if relevantJobs[0].Status.DeleteAfter == nil {
+			relevantJobs[0].Status.DeleteAfter = util.After(time.Now(), *util.ErrDelDelaySeconds)
+			if err := r.Status().Update(ctx, &relevantJobs[0]); err != nil {
+				log.Info("Could not update Delete After for job " + relevantJobs[0].Name)
+			}
+		}
+		if res, err := r.handleJobDeletion(ctx, &relevantJobs[0]); err != nil || res.RequeueAfter > 0 {
+			return res, err
+		}
+	default:
+		log.Error(errors.New("should not reach this point for imagejob"), "imagejob: ", relevantJobs[0])
+	}
+
+	log.Info("done reconcile")
+
+	return ctrl.Result{RequeueAfter: repeatPeriod}, nil
+}
+
+func isNotFound(err error) bool {
+	if err != nil && client.IgnoreNotFound(err) == nil {
+		return true
+	}
+	return false
+}
+
+func (r *Reconciler) handleJobDeletion(ctx context.Context, job *eraserv1alpha1.ImageJob) (ctrl.Result, error) {
+	log.Info("start job deletion")
+	if job.Status.DeleteAfter == nil {
```

Comment:
I think we don't need to check this anymore, since we are checking before we call this method.